src/commands/channels/create.ts

```diff
@@ -0,0 +1,162 @@
+import {eInvalidValue} from '../../utils/errors'
+import * as MixFlags from '../../utils/flags'
+import {DomainOption, validateChannelColor, validateChannelModeOptions} from '../../utils/validations'
+
+const debug = makeDebug('mix:commands:channels:create')
+
+export default class ChannelsCreate extends MixCommand {
+  static description = `create a new channel
+
+Use this command to create a new channel in your Mix project.
+A channel defines a collection of different modalities through
+which users interact with your application. The currently available
+modalities are:
+
+audioscript
+dtmf
+interactivity
+richtext
+tts
+
+Note that, for your convenience, this command will automatically convert
+inputs to the --mode flag as close to the format of the above modes as possible,
+by converting upper-case letters to lower-case and removing deliminating punctuation
+('-' and '_'). So, the modes 'audioscript', 'AUDIO-SCRIPT', and 'a_u-d_i-o----scriPT'
+are all equivalent to the server-side value of 'AUDIO_SCRIPT'. 
+
+The same conversion is done for colours, except that dashes ('-') are replaced 
+with ('_') rather than being deleted. So, 'light-pink', 'light____pink', and 
+'LIGHT_PINK' are also equivalent.
+
+Acceptable channel colors are:
+blue
+brown
+corn-flower
+cyan
+green
+grey
+indigo
+light-green
+light-grey
+light-orange
+light-pink
+light-purple
+orange
+pink
+purple
+ruby
+salmon
+sky
+teal
+yellow
+`
+
+  static examples = [
+    'mix channels:create -P 1922 --name "New IVR channel" --mode ivr',
+  ]
+
+  static flags = {
+    color: flags.string({
+      description: 'channel color',
+    }),
+    mode: MixFlags.modesFlag,
+    name: flags.string({
+      description: 'channel name',
+      required: true,
+    }),
+    project: MixFlags.projectFlag,
+    // output flags
+    json: MixFlags.jsonFlag,
+    yaml: MixFlags.yamlFlag,
+  }
+
+  get domainOptions(): DomainOption[] {
+    debug('get domainOptions()')
+    return ['project']
+  }
+
+  tryDomainOptionsValidation(options: any, domainOptions: DomainOption[]) {
+    debug('tryDomainOptionsValidation()')
+    super.tryDomainOptionsValidation(options, domainOptions)
+
+    if (options.color === undefined) {
+      this.warn(chalk.yellow(`The V4 API currently requires the --color flag to be set.
+This is due to a server-side error in which the default values for these flags are not recognized as valid.
+This invocation of the command will likely fail. If it does not fail, kindly submit an issue on GitHub 
+to let us know.`))
+    } else {
+      try {
+        validateChannelColor(options.color)
+      } catch {
+        throw (eInvalidValue(`Invalid color ${chalk.red(options.color)} supplied.`,
+          [
+            'Check value of --color flag and try again.',
+            `Enter ${chalk.green('mix channels:create help')} to review valid color options.`,
+          ]))
+      }
+    }
+
+    validateChannelModeOptions(options.mode)
+  }
+
```

Comment:
Let's override `setRequestActionMessage` to say

"Creating channel _channelName_ in project _projectID_"

to be consistent with intents:create and entities:create.